src/update_agent/mod.rs

```diff
@@ -35,8 +31,10 @@ enum UpdateAgentState {
     StartState,
     /// Agent initialized.
     Initialized,
-    /// Agent ready to check for updates.
-    Steady,
+    /// Node steady, agent allowed to check for updates.
+    ReportedSteady,
+    /// No further updates available yet.
+    NoNewUpdate,
```

Comment:
Bikeshed: WDYT about naming the `NoNewUpdate` one with the steady keyword since it's likely going to be the stable/steady state in which the agent will mostly sit in? And maybe instead of `ReportedSteady`, something like `ReadyToCheck`? Makes it a little more obvious too that it hasn't checked for updates yet.

Reply:
Unfortunately I'm a dummy so I have already coupled the meaning of "steady" with https://github.com/coreos/zincati/blob/6bad305124522efb494144d107f2b8c18a39bc79/src/fleet_lock/mod.rs#L25

So, to avoid confusion I'd rather stick with the definition of "steady" as "we released the lock, marking the current release green".

Reply:
Hmm, is https://github.com/coreos/airlock/pull/1 still open for change? Would it make sense to change the specs here so that we only actually release the lock once we've also been able to check for an update? It intuitively makes sense to me to do that, but I might be missing something.

Reply:
> Hmm, is coreos/airlock#1 still open for change?

Not really anymore.

> Would it make sense to change the specs here so that we only actually release the lock once we've also been able to check for an update?

I don't think so. We don't want to couple the "did the node reboot" question with the "is Cincinnati-service ok" question inside Zincati. Plus, users can enforce stronger pre-conditions via https://github.com/coreos/zincati/blob/master/dist/systemd/system/zincati.service#L9-L10

Reply:
Ahh OK right. I'd like to chat about this some more, but I don't think it blocks this PR as is.

Reply:
Forwarded to https://github.com/coreos/zincati/issues/239.